### PR TITLE
Fixed ssl connection handling

### PR DESCRIPTION
**Pull request by yasuharu519**
In current implementation, the scope of ssl connection `s` is too big and it is hard to track current connection state, I think.

In this PR, I made ssl connection handle class called `DatadogConnection`. Since that class implemented `__enter__` and `__exit__`, you can use create/close connection with `with` clause.

### CODE REVIEW

**Comment by NBParis** [comment]
Hello @yasuharu519,

Thanks a lot for your contribution and helping us improve this lambda function.
We will review this pull request and let you know if we have any comments.

**Comment by yasuharu519** [comment]
@NBParis 

Thanks! I would fix if you guys found things I fix. 🙏

**Comment by yasuharu519** [comment]
@NBParis 
I fixed conflict 🙏 I would be happy if you review this PR and check, Thanks.

**Comment by NBParis** [comment]
Thanks a lot for this.
Doing some testing on my side. Will come back to you as soon as I get the results.

**Comment by yasuharu519** [comment]
@NBParis I would be happy if there is something I can help testing. (Because longer review time will cause more conflicts I have to fix).

**Comment by NBParis** [comment]
@yasuharu519 So we have done all the testing and this should be good to merge.
We just merged a new improvement on Friday but this should be the latest one. I'll make sure someone from the team approve the PR today.

File: Log/lambda_function.py [hunk 1]
```diff
@@ -8,96 +8,150 @@
...
+            metadata["aws"] = {}
+        aws_meta = metadata["aws"]
+        aws_meta["function_version"] = context.function_version
+        aws_meta["invoked_function_arn"] = context.invoked_function_arn
+        #Add custom tags here by adding new value with the following format "key1:value1, key2:value2"  - might be subject to modifications
+        dd_custom_tags_data = {
+            "forwardername": context.function_name.lower(),
+            "memorysize": context.memory_limit_in_mb
+            }
+        metadata[DD_CUSTOM_TAGS] = ",".join(
```
[the remark this answers is not in the document]
Comment (yasuharu519) [reply Log/lambda_function.py]: The recommend fix looks good. let me fix that way.

**Comment by yasuharu519** [comment]
@achntrl I fixed and resolved conflict. Please review it again.

**Comment by achntrl** [comment]
@yasuharu519 Thank you for your contribution, it is very appreciated!

**Comment by yasuharu519** [comment]
Thank you 👍

**Pull request merged**